test/unit_tests/unittest_scanner_controller.cpp

```diff
@@ -65,182 +68,151 @@ class ScannerControllerTest : public ::testing::Test
+  sendStartReply();
+  start_future.wait_for(std::chrono::seconds(0));
 }
 
-TEST_F(ScannerControllerTest, testHandleStartReplyTimeout)
+TEST_F(ScannerControllerTest, testStartReplyTimeout)
 {
   using ::testing::_;
-  using ::testing::Expectation;
+  using ::testing::InSequence;
+
+  const StartRequest start_request(scanner_config_, 0);
 
   {
-    ::testing::InSequence seq;
-    EXPECT_CALL(scanner_controller_.control_udp_client_, startAsyncReceiving(_, _, _))
-        .WillOnce(::testing::Invoke(
-            [](const ReceiveMode& modi,
-               const TimeoutHandler& timeout_handler,
-               const std::chrono::high_resolution_clock::duration timeout) { timeout_handler("timeout!"); }));
+    InSequence seq;
+    EXPECT_CALL(scanner_controller_.control_udp_client_, startAsyncReceiving(_, _, _)).Times(1);
     EXPECT_CALL(scanner_controller_.data_udp_client_, startAsyncReceiving()).Times(1);
-    EXPECT_CALL(scanner_controller_.control_udp_client_, write(_)).Times(1);
+    EXPECT_CALL(scanner_controller_.control_udp_client_, write(start_request.toRawData())).Times(1);
   }
+
   scanner_controller_.sendStartRequest();
+  simulateUdpTimeout("Udp timeout");
 }
 
-TEST_F(ScannerControllerTest, testHandleStopReplyTimeout)
+TEST_F(ScannerControllerTest, testStop)
 {
   using ::testing::_;
-  using ::testing::Expectation;
+  using ::testing::InSequence;
+
+  const StopRequest stop_request;
 
   {
-    ::testing::InSequence seq;
-    EXPECT_CALL(scanner_controller_.control_udp_client_, startAsyncReceiving(_, _, _))
-        .WillOnce(::testing::Invoke(
-            [](const ReceiveMode& modi,
-               const TimeoutHandler& timeout_handler,
-               const std::chrono::high_resolution_clock::duration timeout) { timeout_handler("timeout!"); }));
-    EXPECT_CALL(scanner_controller_.control_udp_client_, write(_)).Times(1);
+    InSequence seq;
+    EXPECT_CALL(scanner_controller_.control_udp_client_, startAsyncReceiving(_, _, _)).Times(1);
+    EXPECT_CALL(scanner_controller_.control_udp_client_, write(stop_request.toRawData())).Times(1);
   }
-  scanner_controller_.sendStopRequest();
-}
-
-TEST_F(ScannerControllerTest, testHandleErrorNoThrow)
-{
-  ASSERT_NO_THROW(scanner_controller_.handleError("Error Message."));
-}
-
-TEST_F(ScannerControllerTest, testHandleScannerReplyTypeStart)
-{
-  ScannerReplyMsg msg(OP_CODE_START, RES_CODE_ACCEPTED);
-  const auto data{ msg.toRawData() };
-  MaxSizeRawData max_size_data;
-  std::copy_n(data.begin(), data.size(), max_size_data.begin());
 
-  EXPECT_CALL(scanner_controller_.state_machine_, processReplyReceivedEvent(ScannerReplyMsgType::Start)).Times(1);
-  scanner_controller_.handleScannerReply(max_size_data, max_size_data.size());
+  auto stop_future = scanner_controller_.stop();
+  sendStopReply();
+  stop_future.wait_for(std::chrono::seconds(0));
 }
 
-TEST_F(ScannerControllerTest, testHandleScannerReplyTypeStop)
+TEST_F(ScannerControllerTest, testStopReplyTimeout)
```

Comment:
From reading the test I cannot see where a timeout is tested....

Reply:
1. It is expected that receiving is started.... ok.
1. It is expected that a stop request is written... ok.

but nothing about a timeout is expected?

Reply:
I thing the issue is that actually timeout is not the tested behavior but the resend is...

Reply:
Problem is at this point the timeout happens but has no real implementation, we could only test for the log msg and we decided not to do that since the real functionallity is already implemented in another branch, same in testStartReplyTimeout